etp-backend/src/test/clj/solita/etp/test_data/energiatodistus.clj

```diff
@@ -0,0 +1,59 @@
+(ns solita.etp.test-data.energiatodistus
+  (:require [schema.core :as schema]
+            [schema-generators.generators :as g]
+            [flathead.deep :as deep]
+            [solita.common.logic :as logic]
+            [solita.common.schema :as xschema]
+            [solita.etp.test-system :as ts]
+            [solita.etp.test-data.generators :as generators]
+            [solita.etp.schema.common :as common-schema]
+            [solita.etp.schema.energiatodistus :as energiatodistus-schema]
+            [solita.etp.service.energiatodistus :as energiatodistus-service]))
+
+(def generators {schema/Num                   (g/always 1.0M)
+                 common-schema/NonNegative    (g/always 1.0M)
+                 common-schema/IntNonNegative (g/always 1)})
+
+(defn schema-by-version-and-ready-for-signing [versio ready-for-signing?]
+  (cond->> (if (= versio 2013)
+             energiatodistus-schema/EnergiatodistusSave2013
+             energiatodistus-schema/EnergiatodistusSave2018)
+    ready-for-signing? (deep/map-values record?
+                                        (logic/when* xschema/maybe? :schema))))
+
+(defn sisainen-kuorma [versio id]
+  (-> (energiatodistus-service/find-sisaiset-kuormat ts/*db* versio)
+      (->> (filter (comp (partial = id) :kayttotarkoitusluokka-id)))
+      first
+      (dissoc :kayttotarkoitusluokka-id)))
+
+(defn generate [n versio ready-for-signing?]
```

Comment:
erilaiset versiot olisi ehkä parempi olla eri funktioita jotka käyttää luontiin yhteistä funktiota

Reply:
esim. generate-draft, generate-complete-for-signing

Reply:
helpompi käyttää ja helpompi toteuttaa ja helpompi laajentaa uusia versioita

Reply:
Mietin tässä kumpi malli olisi parempi. Toisaalta versiot mukaanlukien erilaisia generate-funktion versioita on jo 4. Yksi tapa voisi olla, että generatesta on versio missä ready-for-signing? on oletuksena false. Tuota valmiiksi allekirjoitettavaa versiota tarvii testeissä tosi harvoin.